packages/shared/src/components/modals/post/boost/BoostPostModal.tsx

```diff
@@ -0,0 +1,281 @@
+import type { ReactElement } from 'react';
+import React, { useState } from 'react';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { Modal } from '../../common/Modal';
+import type { ModalProps } from '../../common/Modal';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../../../typography/Typography';
+import { Button, ButtonSize, ButtonVariant } from '../../../buttons/Button';
+import { useAuthContext } from '../../../../contexts/AuthContext';
+import { CoreIcon, PlusIcon } from '../../../icons';
+import type { Post } from '../../../../graphql/posts';
+import { Image } from '../../../image/Image';
+import { generateQueryKey, RequestKey } from '../../../../lib/query';
+import { Slider } from '../../../fields/Slider';
+import useDebounceFn from '../../../../hooks/useDebounceFn';
+import { largeNumberFormat } from '../../../../lib';
+import { IconSize } from '../../../Icon';
+import { AdsDashboardModal } from './AdsDashboardModal';
+import { postBoostSuccessCover } from '../../../../lib/image';
+import { Origin } from '../../../../lib/log';
+import { BuyCoresModal } from '../../award/BuyCoresModal';
+import { usePostImage } from '../../../../hooks/post/usePostImage';
+import type { TransactionCreated } from '../../../../graphql/njord';
+import { isNullOrUndefined } from '../../../../lib/func';
+
+interface BoostPostModalProps extends ModalProps {
+  post: Post;
+}
+
+const SCREENS = {
+  FORM: 'FORM',
+  BUY_CORES: 'BUY_CORES',
+  DASHBOARD: 'DASHBOARD',
+  SUCCESS: 'SUCCESS',
+} as const;
+
+export type Screens = keyof typeof SCREENS;
+
+export function BoostPostModal({
+  post,
+  ...props
+}: BoostPostModalProps): ReactElement {
+  const { user, updateUser } = useAuthContext();
+  const [activeScreen, setActiveScreen] = useState<Screens>(SCREENS.FORM);
+  const [coresPerDay, setCoresPerDay] = React.useState(5000);
+  const [totalDays, setTotalDays] = React.useState(7);
+  const [queryProps, setQueryProps] = React.useState({
+    coresPerDay,
+    totalDays,
+  });
+  const { data } = useQuery<{ min: number; max: number }>({
+    queryKey: generateQueryKey(
+      RequestKey.PostBoostReach,
+      user,
+      post.id,
+      queryProps.coresPerDay,
+      queryProps.totalDays,
+    ),
+  });
+  const client = useQueryClient();
+  const [debounceSet] = useDebounceFn(setQueryProps, 220);
+  const totalSpendInt = coresPerDay * totalDays;
+  const totalSpend = largeNumberFormat(totalSpendInt);
+  const { mutateAsync: onBoost } = useMutation<{
+    startPostBoost: TransactionCreated;
+  }>({
+    onSuccess: (result) => {
+      setActiveScreen(SCREENS.SUCCESS);
+      const balance = result?.startPostBoost?.balance;
+
+      if (!isNullOrUndefined(balance)) {
+        updateUser({ ...user, balance });
+      }
+
+      // invalidate wallet queries
+      client.invalidateQueries({
+        queryKey: generateQueryKey(RequestKey.Transactions, user),
+        exact: false,
+      });
```

Comment:
Why do you need to invalidate, it happens from feed right, and this only will show in wallet? (not sure) but then you probably don't need to do anything really.